docs/setup-detailed.md

```diff
@@ -1,236 +1,302 @@
+    });
+
+    console.log('🔗 Testing Filecoin JSON-RPC connection...');
+
+    // Get current block
+    const blockNumber = await publicClient.getBlockNumber();
+    console.log(`📦 Current block number: ${blockNumber}`);
+    console.log(`✅ Connected to network: ${filecoinCalibration.name} (Chain ID: ${filecoinCalibration.id})`);
+
+    // Create wallet client
+    const account = privateKeyToAccount(`0x${process.env.PRIVATE_KEY}`);
+    console.log(`👛 Wallet address: ${account.address}`);
+
+    // Check tFIL balance
+    const filBalance = await publicClient.getBalance({ address: account.address });
+    console.log(`💰 tFIL balance: ${formatEther(filBalance)} tFIL`);
+
+    // Test USDFC token balance
+    const usdcBalance = await publicClient.readContract({
+      address: process.env.USDFC_TOKEN_ADDRESS,
+      abi: [
+        {
+          name: 'balanceOf',
+          type: 'function',
+          stateMutability: 'view',
+          inputs: [{ name: 'owner', type: 'address' }],
+          outputs: [{ name: '', type: 'uint256' }],
+        },
+        {
+          name: 'symbol',
+          type: 'function',
+          stateMutability: 'view',
+          inputs: [],
+          outputs: [{ name: '', type: 'string' }],
+        },
+        {
+          name: 'decimals',
+          type: 'function',
+          stateMutability: 'view',
+          inputs: [],
+          outputs: [{ name: '', type: 'uint8' }],
+        },
+      ],
+      functionName: 'balanceOf',
+      args: [account.address],
+    });
+
+    const usdcSymbol = await publicClient.readContract({
+      address: process.env.USDFC_TOKEN_ADDRESS,
+      abi: [{ name: 'symbol', type: 'function', stateMutability: 'view', inputs: [], outputs: [{ name: '', type: 'string' }] }],
+      functionName: 'symbol',
+    });
+
+    console.log(`💵 ${usdcSymbol} balance: ${formatUnits(usdcBalance, 6)} ${usdcSymbol}`);
+
+    console.log('\n🎉 JSON-RPC connection test successful!');
+
+  } catch (error) {
+    console.error('❌ Connection test failed:', error.message);
+    process.exit(1);
+  }
 }
 
-main()
-  .then(() => process.exit(0))
-  .catch(error => {
-    console.error(error);
-    process.exit(1);
-  });
+testConnection();
 ```
 
-Run the verification script:
+### Run the Test
 
 ```bash
-node verify-setup.js
+node test-connection.js
 ```
 
-If everything is set up correctly, you should see output similar to:
-
+**Expected Output:**
 ```
-Connected to Filecoin Calibration Testnet
-Current block number: 1234567
-Next proof set ID: 42
-Maximum proving period: 60 epochs
-Wallet balance: 1.5 tFIL
-Setup verification complete!
+🔗 Testing Filecoin JSON-RPC connection...
+📦 Current block number: 1234567n
+✅ Connected to network: Filecoin Calibration (Chain ID: 314159)
+👛 Wallet address: 0x1234...abcd
+💰 tFIL balance: 50.0 tFIL
+💵 USDFC balance: 30.0 USDFC
+
+🎉 JSON-RPC connection test successful!
 ```
 
-## Step 7: Set Up Development Tools
+## 6. Understanding JSON-RPC Methods
```

Comment:
To me, it's better to link out for details of other methods in the relevant docs, but I like seeing things that are specifically necessary for the Golden Path.